Frontend/src/components/book-catalogue/BookListAssignLoan.tsx

```diff
@@ -0,0 +1,269 @@
+import React, { useEffect, useState } from "react";
```

Comment:
Here you are essentially copying and pastying from `BookListItem`, in any case, I think this isn't necessary. When assigning a loan as borrowed the librarian is supposed to have the physical book in his hands to read the inventory number from. They shouldn't search from a list of all books themselves in this case.

Basically, you can delete this manual search.